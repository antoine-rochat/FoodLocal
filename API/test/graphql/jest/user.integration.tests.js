const { graphql } = require('graphql');
const { makeExecutableSchema } = require('graphql-tools');
const { resolvers, schema: typeDefs } = require('../../../src/graphql/graphqlConfig');
const clearDB = require('../clearDB');
const { populateDB, getTabUsers } = require('../../populateDatabase');

// Making schema graphql
const schema = makeExecutableSchema({ typeDefs, resolvers });

let tabUsers;

const clearAndPopulateDB = async() => {
  // ---------------------------------------- on supprime tout le contenu de la DB ----------------------------------------
  await clearDB();

  // ------------------------------------------- on ajoute le contenu de dÃ©part -------------------------------------------
  await populateDB();

  tabUsers = getTabUsers();
};

describe('Testing graphql resquest user', () => {

  describe('QUERY user', () => {
    // ----------------------users()-------------------------------------- //
    describe('Testing users()', () => {
      let context;
      beforeEach(async() => {
        await clearAndPopulateDB();
        context = { id: tabUsers[0].id, email: tabUsers[0].email, isAdmin: true, kind: tabUsers[0].kind };
      });
      const { query } = {
        query: `
  query {
    users {
      firstname
      lastname
      email
      image
      followingProducers {
        firstname
        lastname
        email
        image
        emailValidated
        phoneNumber
        rating {
          nbRatings
          rating
        }
      }
      emailValidated
      isAdmin
    }
  }`
      };

      it('should get all users', async(done) => {
        const result = await graphql(schema, query, null, context, null);
        expect.assertions(2);
        expect(result.data.users.length).toEqual(tabUsers.length);
        expect(result).toMatchSnapshot();
        done();
      });

      it('should not get all users because not authenticated', async(done) => {
        const result = await graphql(schema, query, null, {}, null);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Sorry, you need to be authenticated to do that.'));
        expect(result).toMatchSnapshot();
        done();
      });

      it('should get all users because not authenticated as administrator', async(done) => {
        context.isAdmin = false;
        const result = await graphql(schema, query, null, context, null);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Sorry, you need to be an administrator to do that.'));
        expect(result).toMatchSnapshot();
        done();
      });
    });

    // ----------------------user(userId: ID!)-------------------------------------- //
    describe('Testing user(userId: ID!)', () => {
      let context;
      beforeEach(async() => {
        await clearAndPopulateDB();
        context = { id: tabUsers[0].id, email: tabUsers[0].email, isAdmin: true, kind: tabUsers[0].kind };
      });

      const { query } = {
        query: `
    query ($id: ID!){
      user(userId: $id) {
        firstname
        lastname
        email
        image
        followingProducers {
          firstname
          lastname
          email
          image
          emailValidated
          phoneNumber
          rating {
            nbRatings
            rating
          }
        }
        emailValidated
        isAdmin
      }
    }`
      };

      it('should get a user by id', async(done) => {
        const variables = { id: tabUsers[0].id };
        const result = await graphql(schema, query, null, context, variables);
        expect.assertions(2);
        expect(result.data.user).not.toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should not get a user because not authenticated', async(done) => {
        const variables = { id: tabUsers[0].id };
        const result = await graphql(schema, query, null, {}, variables);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Sorry, you need to be authenticated to do that.'));
        expect(result).toMatchSnapshot();
        done();
      });

      it('should get a user because not authenticated as administrator', async(done) => {
        context.isAdmin = false;
        const variables = { id: tabUsers[0].id };
        const result = await graphql(schema, query, null, context, variables);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Sorry, you need to be an administrator to do that.'));
        expect(result).toMatchSnapshot();
        done();
      });


      it('should fail getting a user by id because unknown id received', async(done) => {
        context.id = 'abcdefabcdefabcdefabcdef';
        const variables = { id: context.id };
        const result = await graphql(schema, query, null, context, variables);
        expect.assertions(2);
        expect(result.data.user).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should fail getting a user by id because invalid id received (too short)', async(done) => {
        context.id = 'abcdef';
        const variables = { id: context.id };
        const result = await graphql(schema, query, null, context, variables);
        expect.assertions(4);
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual('Received user.id is invalid!');
        expect(result.data.user).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should fail getting a user by id because invalid id received (too long)', async(done) => {
        context.id = 'abcdefabcdefabcdefabcdefabcdef';
        const variables = { id: context.id };
        const result = await graphql(schema, query, null, context, variables);
        expect.assertions(4);
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual('Received user.id is invalid!');
        expect(result.data.user).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });
    });

    // ----------------------updateUser(user: UserInputUpdate!)-------------------------------------- //
    describe('Testing updateUser(user: UserInputUpdate!)', () => {
      let context;
      beforeEach(async() => {
        await clearAndPopulateDB();
        context = { id: tabUsers[0].id, email: tabUsers[0].email, isAdmin: true, kind: tabUsers[0].kind };
      });

      const { mutation } = {
        mutation: `
    mutation ($user: UserInputUpdate!){
  updateUser(user: $user){
    firstname
    lastname
    email
    image
    followingProducers{
      firstname
      lastname
      email
    }
    emailValidated
    isAdmin
  }
}`
      };

      it('should update a user', async(done) => {
        const variables = {
          user: {
            id: tabUsers[0].id,
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(2);
        expect(result.data.updateUser).not.toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should not update a user because missing mendatory information (id)', async(done) => {
        const variables = {
          user: {
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Field value.id of required type ID! was not provided.'));
        expect(result).toMatchSnapshot();
        done();
      });

      it('should not update a user because missing mendatory information (firstname)', async(done) => {
        const variables = {
          user: {
            id: tabUsers[0].id,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(3);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Field value.firstname of required type String! was not provided.'));
        done();
      });

      it('should not update a user because missing mendatory information (lastname)', async(done) => {
        const variables = {
          user: {
            id: tabUsers[0].id,
            firstname: tabUsers[1].firstname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(3);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Field value.lastname of required type String! was not provided.'));
        done();
      });

      it('should not update the user image because image not received (undefined)', async(done) => {
        const variables = {
          user: {
            id: tabUsers[0].id,
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(3);
        expect(result.data.updateUser).not.toBeNull();
        expect(result.data.updateUser.image).toEqual(tabUsers[0].image);
        expect(result).toMatchSnapshot();
        done();
      });

      it('should update the user image to null because null image received', async(done) => {
        const variables = {
          user: {
            id: tabUsers[0].id,
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname,
            image: null
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(3);
        expect(result.data.updateUser).not.toBeNull();
        expect(result.data.updateUser.image).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should not update a user because not authenticated', async(done) => {
        const variables = {
          user: {
            id: tabUsers[0].id,
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, {}, variables);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Sorry, you need to be authenticated to do that.'));
        expect(result).toMatchSnapshot();
        done();
      });

      it('should not update a user because not authenticated as yourself', async(done) => {
        const variables = {
          user: {
            id: tabUsers[1].id,
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('You can\'t modify information of another user than yourself!'));
        expect(result).toMatchSnapshot();
        done();
      });

      it('should fail updating a user because unknown id received', async(done) => {
        context.id = 'abcdefabcdefabcdefabcdef';
        const variables = {
          user: {
            id: context.id,
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(2);
        expect(result.data.updateUser).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should fail updating a user because invalid id received (too short)', async(done) => {
        context.id = 'abcdef';
        const variables = {
          user: {
            id: context.id,
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(4);
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual('Received user.id is invalid!');
        expect(result.data.updateUser).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should fail updating a user because invalid id received (too long)', async(done) => {
        context.id = 'abcdefabcdefabcdefabcdefabcdef';
        const variables = {
          user: {
            id: context.id,
            firstname: tabUsers[1].firstname,
            lastname: tabUsers[1].lastname
          }
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(4);
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual('Received user.id is invalid!');
        expect(result.data.updateUser).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });
    });

    // ----------------------deleteUser(userId: ID!)-------------------------------------- //
    describe('Testing deleteUser(userId: ID!)', () => {
      let context;
      beforeEach(async() => {
        await clearAndPopulateDB();
        context = { id: tabUsers[0].id, email: tabUsers[0].email, isAdmin: true, kind: tabUsers[0].kind };
      });

      const { mutation } = {
        mutation: `
    mutation ($id: ID!){
  deleteUser(userId:$id){
    firstname
    lastname
    email
    image
    followingProducers{
      firstname
      lastname
      email
    }
    emailValidated
    isAdmin
  }
}`
      };

      it('should delete a user', async(done) => {
        const variables = {
          id: tabUsers[0].id
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(2);
        expect(result.data.updateUser).not.toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should not delete a user because not authenticated', async(done) => {
        const variables = {
          id: tabUsers[0].id
        };
        const result = await graphql(schema, mutation, null, {}, variables);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('Sorry, you need to be authenticated to do that.'));
        expect(result).toMatchSnapshot();
        done();
      });

      it('should not delete a user because not authenticated as yourself', async(done) => {
        const variables = {
          id: tabUsers[1].id
        };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(4);
        expect(result.errors).not.toBeNull();
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual(expect.stringContaining('You can\'t modify information of another user than yourself!'));
        expect(result).toMatchSnapshot();
        done();
      });

      it('should fail deleting a user because unknown id received', async(done) => {
        context.id = 'abcdefabcdefabcdefabcdef';
        const variables = { id: context.id };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(2);
        expect(result.data.deleteUser).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should fail deleting a user because invalid id received (too short)', async(done) => {
        context.id = 'abcdef';
        const variables = { id: context.id };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(4);
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual('Received user.id is invalid!');
        expect(result.data.deleteUser).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });

      it('should fail deleting a user because invalid id received (too long)', async(done) => {
        context.id = 'abcdefabcdefabcdefabcdefabcdef';
        const variables = { id: context.id };
        const result = await graphql(schema, mutation, null, context, variables);
        expect.assertions(4);
        expect(result.errors.length).toBe(1);
        expect(result.errors[0].message).toEqual('Received user.id is invalid!');
        expect(result.data.deleteUser).toBeNull();
        expect(result).toMatchSnapshot();
        done();
      });
    });
  });
});
