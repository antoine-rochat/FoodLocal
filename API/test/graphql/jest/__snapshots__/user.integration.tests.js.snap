// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing graphql resquest user MUTATION user Testing deleteUser(userId: ID!) should delete a user 1`] = `
Object {
  "data": Object {
    "deleteUser": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Jérémie",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "firstname": "Antoine",
          "lastname": "Rochaille",
        },
      ],
      "image": "ceci est une image encodée en base64!",
      "isAdmin": false,
      "lastname": "Chaton",
    },
  },
}
`;

exports[`Testing graphql resquest user MUTATION user Testing deleteUser(userId: ID!) should fail deleting a user because invalid id received (too long) 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing deleteUser(userId: ID!) should fail deleting a user because invalid id received (too short) 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing deleteUser(userId: ID!) should fail deleting a user because unknown id received 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
}
`;

exports[`Testing graphql resquest user MUTATION user Testing deleteUser(userId: ID!) should not delete a user because not authenticated 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": Array [
    [GraphQLError: Sorry, you need to be authenticated to do that.],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing deleteUser(userId: ID!) should not delete a user because not authenticated as yourself 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": Array [
    [GraphQLError: You can't modify information of another user than yourself!],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should fail updating a user because invalid id received (too long) 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should fail updating a user because invalid id received (too short) 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should fail updating a user because unknown id received 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: The received id is not in the database!],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should not update a user because missing mendatory information (id) 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$user" got invalid value { firstname: "Loyse", lastname: "Hug" }; Field value.id of required type ID! was not provided.],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should not update a user because not authenticated 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: Sorry, you need to be authenticated to do that.],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should not update a user because not authenticated as yourself 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: You can't modify information of another user than yourself!],
  ],
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should not update the user image because image not received (undefined) 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Loyse",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "firstname": "Antoine",
          "lastname": "Rochaille",
        },
      ],
      "image": "ceci est une image encodée en base64!",
      "isAdmin": false,
      "lastname": "Hug",
    },
  },
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should update a user 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Loyse",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "firstname": "Antoine",
          "lastname": "Rochaille",
        },
      ],
      "image": "ceci est une image encodée en base64!",
      "isAdmin": false,
      "lastname": "Hug",
    },
  },
}
`;

exports[`Testing graphql resquest user MUTATION user Testing updateUser(user: UserInputUpdate!) should update the user image to null because null image received 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Loyse",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "firstname": "Antoine",
          "lastname": "Rochaille",
        },
      ],
      "image": null,
      "isAdmin": false,
      "lastname": "Hug",
    },
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing deleteUser(userId: ID!) should delete a user 1`] = `
Object {
  "data": Object {
    "deleteUser": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Jérémie",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "firstname": "Antoine",
          "lastname": "Rochaille",
        },
      ],
      "image": "ceci est une image encodée en base64!",
      "isAdmin": false,
      "lastname": "Chaton",
    },
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing deleteUser(userId: ID!) should fail deleting a user because invalid id received (too long) 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing deleteUser(userId: ID!) should fail deleting a user because invalid id received (too short) 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing deleteUser(userId: ID!) should fail deleting a user because unknown id received 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing deleteUser(userId: ID!) should not delete a user because not authenticated 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": Array [
    [GraphQLError: Sorry, you need to be authenticated to do that.],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing deleteUser(userId: ID!) should not delete a user because not authenticated as yourself 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": Array [
    [GraphQLError: You can't modify information of another user than yourself!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should fail updating a user because invalid id received (too long) 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should fail updating a user because invalid id received (too short) 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should fail updating a user because unknown id received 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: The received id is not in the database!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should not update a user because missing mendatory information (firstname) 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$user" got invalid value { id: "5c2b6e63379d9d960e69b4ab", lastname: "Hug" }; Field value.firstname of required type String! was not provided.],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should not update a user because missing mendatory information (id) 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$user" got invalid value { firstname: "Loyse", lastname: "Hug" }; Field value.id of required type ID! was not provided.],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should not update a user because missing mendatory information (lastname) 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$user" got invalid value { id: "5c2b6e63379d9d960e69b51b", firstname: "Loyse" }; Field value.lastname of required type String! was not provided.],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should not update a user because not authenticated 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: Sorry, you need to be authenticated to do that.],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should not update a user because not authenticated as yourself 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: You can't modify information of another user than yourself!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should not update the user image because image not received (undefined) 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Loyse",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "firstname": "Antoine",
          "lastname": "Rochaille",
        },
      ],
      "image": "ceci est une image encodée en base64!",
      "isAdmin": false,
      "lastname": "Hug",
    },
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should update a user 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Loyse",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "firstname": "Antoine",
          "lastname": "Rochaille",
        },
      ],
      "image": "ceci est une image encodée en base64!",
      "isAdmin": false,
      "lastname": "Hug",
    },
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing updateUser(user: UserInputUpdate!) should update the user image to null because null image received 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Loyse",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "firstname": "Antoine",
          "lastname": "Rochaille",
        },
      ],
      "image": null,
      "isAdmin": false,
      "lastname": "Hug",
    },
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing user(userId: ID!) should fail getting a user by id because invalid id received (too long) 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing user(userId: ID!) should fail getting a user by id because invalid id received (too short) 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": Array [
    [GraphQLError: Received user.id is invalid!],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing user(userId: ID!) should fail getting a user by id because unknown id received 1`] = `
Object {
  "data": Object {
    "user": null,
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing user(userId: ID!) should get a user because not authenticated as administrator 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": Array [
    [GraphQLError: Sorry, you need to be an administrator to do that.],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing user(userId: ID!) should get a user by id 1`] = `
Object {
  "data": Object {
    "user": Object {
      "email": "jerem@user.ch",
      "emailValidated": false,
      "firstname": "Jérémie",
      "followingProducers": Array [
        Object {
          "email": "antoine@paysan.ch",
          "emailValidated": false,
          "firstname": "Antoine",
          "image": "Ceci est une image encodée en base64!",
          "lastname": "Rochaille",
          "phoneNumber": "0761435196",
          "rating": Object {
            "nbRatings": 3,
            "rating": 3.3333333333333335,
          },
        },
      ],
      "image": "ceci est une image encodée en base64!",
      "isAdmin": false,
      "lastname": "Chaton",
    },
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing user(userId: ID!) should not get a user because not authenticated 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": Array [
    [GraphQLError: Sorry, you need to be authenticated to do that.],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing users() should get all users 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "email": "jerem@user.ch",
        "emailValidated": false,
        "firstname": "Jérémie",
        "followingProducers": Array [
          Object {
            "email": "antoine@paysan.ch",
            "emailValidated": false,
            "firstname": "Antoine",
            "image": "Ceci est une image encodée en base64!",
            "lastname": "Rochaille",
            "phoneNumber": "0761435196",
            "rating": Object {
              "nbRatings": 3,
              "rating": 3.3333333333333335,
            },
          },
        ],
        "image": "ceci est une image encodée en base64!",
        "isAdmin": false,
        "lastname": "Chaton",
      },
      Object {
        "email": "loyse@user.ch",
        "emailValidated": false,
        "firstname": "Loyse",
        "followingProducers": Array [
          Object {
            "email": "antoine@paysan.ch",
            "emailValidated": false,
            "firstname": "Antoine",
            "image": "Ceci est une image encodée en base64!",
            "lastname": "Rochaille",
            "phoneNumber": "0761435196",
            "rating": Object {
              "nbRatings": 3,
              "rating": 3.3333333333333335,
            },
          },
        ],
        "image": null,
        "isAdmin": false,
        "lastname": "Hug",
      },
    ],
  },
}
`;

exports[`Testing graphql resquest user QUERY user Testing users() should get all users because not authenticated as administrator 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Sorry, you need to be an administrator to do that.],
  ],
}
`;

exports[`Testing graphql resquest user QUERY user Testing users() should not get all users because not authenticated 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Sorry, you need to be authenticated to do that.],
  ],
}
`;
